packages/logfire/src/logfireConfig.ts

```diff
@@ -161,8 +165,8 @@ export function configure(config: LogfireConfigOptions = {}) {
     logfireApi.configureLogfireApi({ otelScope })
   }
 
-  const baseUrl = resolveBaseUrl(cnf.advanced?.baseUrl)
   const token = cnf.token ?? env.LOGFIRE_TOKEN
+  const baseUrl = resolveBaseUrl(cnf.advanced?.baseUrl, token)
```

Comment:
Maybe instead what can be done:
- first check for `resolveSendToLogfire(cnf.sendToLogfire, token)`. If `false`, set `baseUrl` to an empty string (as it shouldn't be used).
- If `true`, assert that `token` is not undefined, and fail eagerly.
- Only then call `resolveBaseUrl()`

This way you can type the `token` parameter as `string` in `resolveBaseUrl()` and `getBaseUrlFromToken()`